src/main/java/com/shaneschulte/mc/clanarena/commands/CreateCmd.java

```diff
@@ -104,252 +80,98 @@ else if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
-                    updateProgressBossBar(player.getName());
-                    return;
-                }
-
-                // if there are already spawns being set, add to the list
-                ArrayList<Location> newList = currentTeam1SpawnsByPlayer.get(player.getName());
-                newList.add(player.getLocation());
-                currentTeam1SpawnsByPlayer.replace(player.getName(), newList);
-                updateProgressBossBar(player.getName());
-
-                if (currentTeam1SpawnsByPlayer.get(player.getName()).size() > minSpawns -1) MsgUtils.sendMessage(player, MsgUtils.Colors.VARIABLE + "" +
-                    currentTeam1SpawnsByPlayer.get(player.getName()).size() + MsgUtils.Colors.INFO + " spawns for team1. Type \"" + MsgUtils.Colors.SUCCESS + "next" +
-                    MsgUtils.Colors.INFO + "\" to move on");
+            if (getSetupArena(p).getArenaState().equals(ArenaSetupState.SELECTING_TEAM1_SPAWNS)) {
+
+                // add to the list
+                getSetupArena(p).addFirstTeamSpawn(p.getLocation());
             }
 
             // if team 02 spawns
             else {
-                // if no spawns set yet, add the first spawn and exit
-                if (!(currentTeam2SpawnsByPlayer.containsKey(player.getName()))) {
-                    ArrayList<Location> newList = new ArrayList<>();
-                    newList.add(player.getLocation());
-                    currentTeam2SpawnsByPlayer.put(player.getName(), newList);
-                    updateProgressBossBar(player.getName());
-                    return;
-                }
-
-                // if there are already spawns being set, add to the list
-                ArrayList<Location> newList = currentTeam2SpawnsByPlayer.get(player.getName());
-                newList.add(player.getLocation());
-                currentTeam2SpawnsByPlayer.replace(player.getName(), newList);
-                updateProgressBossBar(player.getName());
-
-                if (currentTeam2SpawnsByPlayer.get(player.getName()).size() > minSpawns -1) MsgUtils.sendMessage(player, MsgUtils.Colors.VARIABLE + "" +
-                        currentTeam2SpawnsByPlayer.get(player.getName()).size() + MsgUtils.Colors.INFO + " spawns for team1. Type \"" + MsgUtils.Colors.SUCCESS + "done" +
-                        MsgUtils.Colors.INFO + "\" to move on");
+
+                // add to the list
+                getSetupArena(p).addSecondTeamSpawn(p.getLocation());
             }
         }
 
-        else if (playersInTheMiddleOfSelectingSpawnPoints.contains(player.getName()) && !(currentTeam2SpawnsByPlayer.containsKey(player.getName())) && !(playersAddingSecondTeamSpawnPoints.contains(player.getName())) && event.getMessage().equalsIgnoreCase("next")) {
+        else if (event.getMessage().equalsIgnoreCase("next") && getSetupArena(p).getArenaState().equals(ArenaSetupState.SELECTING_TEAM1_SPAWNS) && getSetupArena(p).canMoveOnFromSettingFirstTeamSpawns()) {
```

Comment:
you should move more of this logic into SetupArena class, rather than using it as a glorified variable